scripts/o3de/ExportScripts/export_source_ios_xcode.py

```diff
@@ -44,51 +64,81 @@ def export_ios_xcode_project(ctx: exp.O3DEScriptExportContext,
         logger = logging.getLogger()
         logger.setLevel(logging.ERROR)
 
-    tools_build_folder_str = str(tools_build_folder)
-    ios_build_folder_str = str(ios_build_folder)
+    default_base_path =  ctx.project_path
+
+    if not tools_build_path:
+        tools_build_path = default_base_path / 'build/tools'
+    elif not tools_build_path.is_absolute():
+        tools_build_path = default_base_path / tools_build_path
+
+    tools_build_path_str = str(tools_build_path)
+    ios_build_folder_str = str(target_ios_project_path)
+
+    # Resolve (if possible) and validate any provided seedlist files
+    validated_seedslist_paths = exp.validate_project_artifact_paths(project_path=ctx.project_path,
+                                                                    artifact_paths=seedlist_paths)
+    
+    # Preprocess the seed file paths in case there are any wildcard patterns to use
+    preprocessed_seedfile_paths = exp.preprocess_seed_path_list(project_path=ctx.project_path,
+                                                                paths=seedfile_paths)
+    
+    eutil.process_level_names(ctx, preprocessed_seedfile_paths, level_names, ASSET_PLATFORM)
 
     # Optionally build the toolchain needed to process the assets
+    # note: For now, I am keeping the tool building logic restricted on iOS, b/c atm it's unreliable for most configurations besides this one
     if should_build_tools:
-        exp.process_command(["cmake", "-B", tools_build_folder_str, '-G', "Xcode"], cwd=ctx.project_path)
+        exp.process_command(["cmake", "-B", tools_build_path_str, '-G', "Xcode"], cwd=ctx.project_path)
 
-        exp.process_command(["cmake", "--build", tools_build_folder_str, "--target", "AssetProcessorBatch", "AssetBundlerBatch", "--config", "profile"],
+        exp.process_command(["cmake", "--build", tools_build_path_str, "--target", "Editor", "AssetProcessor", "AssetBundler","AssetProcessorBatch", "AssetBundlerBatch", "--config", "profile"],
                     cwd=ctx.project_path)
         
     # Optionally process the assets
-    if not skip_asset_processing:
-        asset_processor_batch_path = exp.get_asset_processor_batch_path(tools_build_folder, required=True)
-        exp.process_command([ str(asset_processor_batch_path), '--platforms=ios',
-                        '--project-path', ctx.project_path ], cwd=ctx.project_path)
+    eutil.build_and_bundle_assets(ctx, should_build_all_assets, tools_build_path, False, 
+                                  tool_config, False, fail_on_asset_errors,
+                                  [ASSET_PLATFORM], validated_seedslist_paths, preprocessed_seedfile_paths, 
+                                  ctx.project_path/'AssetBundling', max_bundle_size,
+                                  logger)
+
+    if should_build_all_assets:
+        if (ctx.project_path / 'AssetBundling/Bundles').is_dir():
+            (ctx.project_path / 'Pak' / f'{ctx.project_name}_ios_paks').mkdir(parents=True)
```

Comment:
Is 'Pak' the final fixed path we want to use? shouldn't this be in the build folder instead in case they have a folder called 'Pak' for something else at the project root level. And if we do want to use the root 'Pak' (which I would not recommend), it would need to go into the .git_ignore file otherwise they will see this in a git changed file.

Reply:
Also, `ctx.project_path / 'Pak'` is repeated a lot here, should go into a variable to re-use